spec/configure_helper_spec.rb

```diff
@@ -1,18 +1,41 @@
 require 'spec_helper.rb'
 
+shared_context 'with temp dir' do
+  before do
+    @tmpdir_path = Dir.mktmpdir
+    @pwd_before_spec_run = Dir.pwd
+    Dir.chdir(tmpdir_path)
+  end
+
+  after do
+    Dir.chdir(pwd_before_spec_run)
+    puts "tmp path is #{tmpdir_path}"
+    FileUtils.rm_rf(tmpdir_path)
+  end
+
+  attr_reader :tmpdir_path
+  attr_reader :pwd_before_spec_run
+end
+
 describe Fastlane::Helper::ConfigureHelper do
+  include_context 'with temp dir'
 
   describe '#add_file' do
     let(:destination) { 'path/to/destination' }
 
+    # rubocop:disable RSpec/ExampleLength
     it 'shows the user an error when the destination is not ignored in Git' do
       allow(Fastlane::Helper::GitHelper).to receive(:is_ignored?)
         .with(path: destination)
         .and_return(false)
 
+      allow(Fastlane::Helper::FilesystemHelper).to receive(:project_path)
+        .and_return(Pathname.new(tmpdir_path))
```

Comment:
Under the hood, if you follow the chain of calls, you'll eventually get to this method which traverse the caller's parent folders graph (list, tree? 🤔 ) backwards till it find one which contains a repo and identifies it as the `project_path`.

In the context of a temp folder that breaks apart, of course and the tests end up stuck.

Reply:
🤔 I should probably make this a comment in the test, actually...

Reply:
Why not run `git init` inside the tmpdir to make it a git repo? A step which you could add as part of the `in_tmp_dir` scoper/helper I suggested in my previous comment for example.